### ISSUE TITLE
Make MultiElevationProvider configureable

### ISSUE THREAD

**Issue by boldtrn** [graphhopper/graphhopper#1243]
As mentioned in #1228.  I think we should make it possible to configure the providers used in the MultiElevationProvider. Currently for the global provider there is only the GMTED provider, but the SRTM provider could be specified as srtm, cgiar, or srtmgl1. In the future the global provider could be changed to ALOS data or a similar provider.
 - `graph.elevation.multi.srtm=cgiar`
 - `graph.elevation.multi.global=gmted`
 - `graph.elevation.multi.srtm.cache_dir=~/srtm/`
 - `graph.elevation.multi.global.cache_dir=~/gmted/`

This follows the naming scheme I proposed [here](https://github.com/graphhopper/graphhopper/pull/1235/files#diff-10a05bb2835502ef4ab61dbc99bba02eR30).

**Comment by karussell** [comment]
wouldn't it make more sense to define just the order like:

graph.elevation.provider=gmted,cgiar

and then the different cache dirs?

`graph.elevation.<provider>.cache_dir=/tmp/srtm/`

This way it would be easy to add a third provider and keep cache dir config

**Comment by boldtrn** [comment]
Ok sure, how about we mix both approaches?

I think we should still configure the multi elevation provider like:
```
graph.elevation.provider=multi
graph.elevation.multi.srtm=cgiar
graph.elevation.multi.global=gmted
```

This makes it easier to understand what is happening I think. Also if we think about creating a more complicated setup using even more sources, maybe some very high precision data for some regions, etc.

But yes specifying the cache dirs as you mentioned makes a lot of sense to me.
```
graph.elevation.<provider>.cache_dir=/tmp/srtm/
```

**Comment by karussell** [comment]
I find this line
graph.elevation.multi.srtm=cgiar
a bit hard to understand. What does srtm in this context mean ... also "multi" in this case is limited to "dual elevation provider"

**Comment by boldtrn** [comment]
> What does srtm in this context mean

This means the elevation provider that covers the SRTM area.

> in this case is limited to "dual elevation provider"

Yes but we could easily add new lines, like `graph.elevation.multi.alps=alps` (assuming we might have a high precision elevation provider for the alps). And there might be other providers as well. But maybe it is easier for these special cases to simply have a switch `graph.elevation.multi.alps=true|false`.

**Comment by karussell** [comment]
I think we should define order of access instead of "global" or "srtm". From outside such an order would be also applicable in a local context like 1. use high precision data and 2. if not available use normal srtm and 3. if not available use global data. With this explanation the line
`graph.elevation.provider=highprecision,cgiar,gmted` would be relative easy to understand IMO and more generic, also the order can be defined by the user and is not fixed in code.

**Comment by karussell** [comment]
Furthermore when we add a new elevation provider we do not need to change the multi elevation provider with this more generic approach.

**Comment by boldtrn** [comment]
> I think we should define order of access instead of "global" or "srtm". From outside such an order would be also applicable in a local context like 1. use high precision data and 2. if not available use normal srtm and 3. if not available use global data.

True, this could be done. Should I change the MultiElevationProvider accordingly?

**Comment by karussell** [comment]
Ok, but not high prio :)
